integration-tests/scripts/baseline/main.py

```diff
@@ -46,22 +46,78 @@
 # in the empty document to be explicit.
 EMPTY_BASELINE_STRUCTURE = []
 
+# In which format we expect to find dates in the baseline files
+DATE_FORMAT = "%Y-%m-%d %H:%M:%S"
+
+
+def object_parser_hook(data):
+    result = data
+
+    for f in (datetime_parser, memory_parser):
+        result = f(result)
+
+    return result
+
+
+def memory_parser(data):
+    result = {}
+    multipliers = {
+        "GiB": 1024,
+        "MiB": 1,
+        "KiB": 1 / 1024,
+    }
+
+    for k, v in data.items():
+        if k in ('Mem') and isinstance(v, str) and len(v) > 3:
+            suffix = v[-3:]
+            multiplier = multipliers.get(suffix, 0)
+
+            if multiplier == 0:
+                result[k] = 0
+            else:
+                result[k] = float(v[:-3]) * multiplier
+        else:
+            result[k] = v
```

Comment:
Is there any chance of nested data that also contains memory fields?

Reply:
My understanding is that the object_hook will be invoked for every nested object, but I can verify.

Reply:
Yep, verified, it's applied to the nested objects as well.